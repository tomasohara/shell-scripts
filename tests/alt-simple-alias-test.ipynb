{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Alt. simple alias test\n",
    "# uses multiple commands per cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n",
      "1\n"
     ]
    }
   ],
   "source": [
    "alias count-words='wc -w'\n",
    "alias count-lines='wc -l'\n",
    "echo abc def ght | count-words\n",
    "echo abc def ght | count-lines"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
