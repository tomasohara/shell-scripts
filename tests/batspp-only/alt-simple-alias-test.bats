#!/usr/bin/env bats
#
# This test file was generated using Batspp
# https://github.com/LimaBD/batspp
#

# Constants
VERBOSE_DEBUG=""
TEMP_DIR="/tmp/batspp-107003"

# Setup function
# $1 -> test name
function run_setup () {
	test_folder=$(echo $TEMP_DIR/$1-$$)
	mkdir --parents "$test_folder"
	cd "$test_folder" || echo Warning: Unable to "cd $test_folder"
}

# Teardown function
function run_teardown () {
	: # Nothing here...
}

@test "test of line 4" {
	run_setup "test-of-line-4"

	# Assertion of line 7
	alias count-words='wc -w'
	alias count-lines='wc -l'
	echo abc def ght | count-words
	shopt -s expand_aliases
	print_debug "$(echo abc def ght | count-lines)" "$(echo -e '3\n1\n')"
	[ "$(echo abc def ght | count-lines)" == "$(echo -e '3\n1\n')" ]

	run_teardown
}

# This prints debug data when an assertion fail
# $1 -> actual value
# $2 -> expected value
function print_debug() {
	echo "=======  actual  ======="
	bash -c "echo \"$1\" $VERBOSE_DEBUG"
	echo "======= expected ======="
	bash -c "echo \"$2\" $VERBOSE_DEBUG"
	echo "========================"
}
