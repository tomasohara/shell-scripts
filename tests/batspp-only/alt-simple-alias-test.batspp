# Alt. simple alias test
# uses multiple commands per cell

$ alias count-words='wc -w'
$ alias count-lines='wc -l'
$ echo abc def ght | count-words
$ echo abc def ght | count-lines
3
1
