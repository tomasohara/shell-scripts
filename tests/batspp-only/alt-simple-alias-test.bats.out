1..1
not ok 1 test-1
# (in test file batspp-only/alt-simple-alias-test.bats, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== alias count-words='wc -w'\n ==========
# ========= $ alias count-lines='wc -l'\n$ echo abc def ght | count-words\n$ echo abc def ght | count-lines\n3\n1 =========
# $ alias count-lines='wc -l'
# $ echo abc def ght | count-words
# $ echo abc def ght | count-lines
# 3
# 1
# ============================
